File: code-external/Accuracy_test.cpp
void evaluatePerformance(vector<vector<double>>& testInputs, vector<vector<double>>& testTargets) {
        double correctPredictions = 0.0;

        for (size_t i = 0; i < testInputs.size(); ++i) {
            vector<double>& input = testInputs[i];
            vector<double>& target = testTargets[i];

            vector<double> prediction = predict(input);

            // Assuming a binary classification problem
            if ((prediction[0] > 0.5 && target[0] > 0.5) || (prediction[0] <= 0.5 && target[0] <= 0.5)) {
                correctPredictions += 1.0;
            }
        }

        double accuracy = correctPredictions / testInputs.size();
        cout << "Accuracy on Test Dataset: " << accuracy * 100.0 << "%\n";
    }

int main(){
    neural_network.evaluatePerformance(training_inputs, training_targets);
}

File: code-external/Accuracy_test.rs
fn evaluate_performance(&self, test_inputs: &Vec<Vec<f64>>, test_targets: &Vec<Vec<f64>>) {
    let predictions = self.predict(test_inputs);
    let accuracy = calculate_accuracy(&predictions, test_targets);

    println!("Accuracy on Test Dataset: {:.2}%", accuracy * 100.0);
}

fn calculate_accuracy(predictions: &Vec<Vec<f64>>, targets: &Vec<Vec<f64>>) -> f64 {
    let correct_predictions = predictions
        .iter()
        .zip(targets.iter())
        .filter(|&(pred, target)| (pred[0] > 0.5 && target[0] > 0.5) || (pred[0] <= 0.5 && target[0] <= 0.5))
        .count() as f64;

    correct_predictions / predictions.len() as f64
}

int main(){
    neural_network.evaluate_performance(&test_inputs, &training_targets);
}